Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contest/ContestDetailsPage.tsx

```diff
@@ -64,6 +76,26 @@ const ContestDetailsPage = () => {
         [ isOfficial, contestDetails?.participantsCountByContestType ],
     );
 
+    const canAccessCompeteButton = useMemo(
+        () => contestDetails?.canBeCompeted || contestDetails?.isAdminOrLecturerInContest,
+        [ contestDetails ],
+    );
+
+    const canAccessPracticeButton = useMemo(
+        () => contestDetails?.canBePracticed || contestDetails?.isAdminOrLecturerInContest,
+        [ contestDetails ],
+    );
+
+    const canOnlyAdminCompete = useMemo(
+        () => !contestDetails?.canBeCompeted && contestDetails?.isAdminOrLecturerInContest,
+        [ contestDetails ],
+    );
+
+    const canOnlyAdminPractice = useMemo(
+        () => !contestDetails?.canBePracticed && canAccessPracticeButton,
+        [ canAccessPracticeButton, contestDetails?.canBePracticed ],
+    );
+
```

Comment:
I think here can be simplified to only one function.

Reply:
modified.